mountpoint-s3/src/prefetch/part.rs

```diff
@@ -11,17 +15,28 @@ pub struct Part {
     key: String,
     offset: u64,
     bytes: Bytes,
+    checksum: ChecksumCrc,
 }
 
 impl Part {
     pub fn new(key: &str, offset: u64, bytes: Bytes) -> Self {
+        let checksum = checksums_crc32c(&bytes, None);
```

Comment:
Maybe make the caller pass in a checksum rather than computing it here.

Reply:
Actually I can't tell if you're intentionally computing it both here and in the caller, but that doesn't seem necessary.

Reply:
I intentionally computing it so that the caller can validate its own checksum with the one in the part struct, but it's probably not necessary.